samples/extensions/full_screen_exclusive/full_screen_exclusive.cpp

```diff
@@ -0,0 +1,1177 @@
+						needs_recreate                                             = true;
+						is_full_screen_exclusive                                   = true;
+						LOGI("Exclusive Fullscreen Mode Detected!")
+					}
+					break;
+				default:        // FullscreenExclusiveEXT = Default
+					needs_recreate = false;
+					break;
+			}
+
+			// now if to recreate the swapchain and everything related:
+			if (needs_recreate)
+			{
+				FullScreenExclusive::recreate();
+			}
+		}
+	}
+}
+
+void FullScreenExclusive::update_application_window()
+{
+	if (application_window_status == ApplicationWindowMode::Fullscreen && is_windowed)        // check if it is already in fullscreen, if is, then do nothing
+	{
+		is_windowed = false;
+
+		if (HWND_style == 0)
+			HWND_style = GetWindowLong(HWND_application_window, GWL_STYLE);
+		if (HWND_extended_style == 0)
+			HWND_extended_style = GetWindowLong(HWND_application_window, GWL_EXSTYLE);
+
+		long HWND_newStyle          = HWND_style & (~WS_BORDER) & (~WS_DLGFRAME) & (~WS_THICKFRAME);
+		long HWND_extended_newStyle = HWND_extended_style & (~WS_EX_WINDOWEDGE);
+
+		SetWindowLong(HWND_application_window, GWL_STYLE, HWND_newStyle | static_cast<long>(WS_POPUP));
+		SetWindowLong(HWND_application_window, GWL_EXSTYLE, HWND_extended_newStyle | WS_EX_TOPMOST);
+
+		ShowWindow(HWND_application_window, SW_SHOWMAXIMIZED);
+	}
+	else if (application_window_status == ApplicationWindowMode::Windowed && !is_windowed)        // check if it is already "windowed", if is, then do nothing
+	{
+		is_windowed = true;
+
+		SetWindowLong(HWND_application_window, GWL_STYLE, HWND_style);
+		SetWindowLong(HWND_application_window, GWL_EXSTYLE, HWND_extended_style);
+		ShowWindow(HWND_application_window, SW_SHOWNORMAL);
+		SetWindowPlacement(HWND_application_window, &wpc);
+	}
+}
+
+void FullScreenExclusive::recreate()
+{
+	// Check if there IS a device, if not don't do anything
+	if (context.device != VK_NULL_HANDLE)
+	{
+		// Step: 0) Idle the device, destroy/teardown the current swapchain and frame buffers.
+		vkDeviceWaitIdle(context.device);        // pause the renderer
+		teardown_framebuffers();                 // basically destroy everything swapchain related
+
+		// Step: 1) recreate the swapchain with its properly selected FullscreenExclusive enum value
+		init_swapchain();
+
+		// Step: 2) recreate the frame buffers using the newly created swapchain
+		init_framebuffers();
+
+		// Step: 3-1) update the window mode, corresponding to the FullscreenExclusive enum value
+		update_application_window();
+
+		// Step: 3-2) remember: ALWAYS change the application window mode BEFORE acquire the full screen exclusive mode!
+		if (is_full_screen_exclusive)
+		{
+			VkResult result = vkAcquireFullScreenExclusiveModeEXT(context.device, context.swapchain);
+			LOGI("vkAcquireFullScreenExclusiveModeEXT: " + vkb::to_string(result))        // it would be necessary to learn the result on an unsuccessful attempt
```

Comment:
You should only call `vkb::to_string(result)` if the result is not `VK_SUCCESS`.
Otherwise, you always get "UNKNOWN_ERROR" as output!
Or extend `vkb::to_string` to also handle that error code.